src/main/java/fr/techad/edc/client/internal/io/HttpReaderImpl.java

```diff
@@ -101,6 +96,17 @@ public HttpReaderImpl(HttpClient client, ClientConfiguration clientConfiguration
         return labels;
     }
 
+    @Override
+    public I18NContent readLabel(Set<String> languageCodes) throws IOException, InvalidUrlException {
+        I18NContent i18NContent = null;
+        if (languageCodes != null) {
+            for (String languageCode : languageCodes) {
+                i18NContent = readi18NContent(languageCode);
```

Comment:
Are you sure?
We get only the last i18n labels and errors.

Need to review this part.

Add unit test to validate your source code.